planner/core/exhaust_physical_plans.go

```diff
@@ -1247,7 +1247,7 @@ func (cwc *ColWithCmpFuncManager) BuildRangesByRow(ctx sessionctx.Context, row c
 		}
 		exprs = append(exprs, newExpr) // nozero
 	}
-	ranges, err := ranger.BuildColumnRange(exprs, ctx, cwc.TargetCol.RetType, cwc.colLength)
+	ranges, _, _, err := ranger.BuildColumnRange(exprs, ctx, cwc.TargetCol.RetType, cwc.colLength, 0)
```

Comment:
Need to add a `todo` comment here?

Reply:
Notice that we don't restrict the mem limit when building column ranges here, so its ok to ignore the second and the third return values. The second return value must be the same as `exprs` and the third must be `nil`. `BuildRangesByRow` is called when we refill inner ranges when executing IndexJoin. At that time we don't need to restrict the mem limit of ranges.

Reply:
I Add comment in code.